crates/transaction-pool/src/maintain/interfaces.rs

```diff
@@ -0,0 +1,253 @@
+//! Interfaces for the transaction pool maintenance components
+//!
+//! This module defines traits for the key components in the transaction pool
+//! maintenance system, allowing them to be replaced with custom implementations.
+
+use crate::{
+    blobstore::BlobStoreUpdates,
+    maintain::{
+        canon_processor::CanonEventProcessor,
+        drift_monitor::{DriftMonitor, DriftMonitorResult, PoolDriftState},
+    },
+    traits::TransactionPoolExt,
+    PoolTransaction,
+};
+use alloy_primitives::{Address, BlockHash, BlockNumber};
+use futures_util::Future;
+use reth_chain_state::CanonStateNotification;
+use reth_chainspec::ChainSpecProvider;
+use reth_primitives_traits::NodePrimitives;
+use reth_storage_api::{BlockReaderIdExt, StateProviderFactory};
+use std::{collections::HashSet, pin::Pin};
+
+/// Trait defining the interface for a drift monitor
+pub trait DriftMonitoring<Client>: Future<Output = DriftMonitorResult> + Send {
```

Comment:
I find this trait a bit too invasive, I feel like this would enshrine a certain behaviour for the maintain loop, which isn't exactly what we want here.

can we take one step back here and focus on encapsulating standalone logic, so that a user can still write a totally custom maintain logic, but with less verbosity.
rn all of this is now part of the trait, and the user is forced to re-implement this completely?

I like the way the event handlers are now extracted, so maybe we start a bit smaller and start with helpers for just handling the  CanonStateNotification?